### ISSUE TITLE
fromBetweenVectors() giving wrong answer

### ISSUE THREAD

**Issue by esromneb** [infusion/Quaternion.js#13]
When I edit the test `should rotate one vector onto the other` to use these values:
```
    var u = [0,0,1];
    var v = [0,0,-1];

```

I get this failure:
```
  1 failing

  1) Quaternions
       should rotate one vector onto the other:

      AssertionError [ERR_ASSERTION]: '-k' == 'k'
      + expected - actual

      --k
      +k
      
      at Function.assert.q (tests/quaternion.test.js:38:12)
      at Context.<anonymous> (tests/quaternion.test.js:713:12)
      at processImmediate (internal/timers.js:464:21)
```

Unless I'm missing something, this should be a simple rotation of 180 degrees (represented as a Quaternion of course). Is this some sort of edge condition?

**Comment by infusion** [comment]
Good catch on this and thanks for bringing this up. This is a known edge case that must be handled somehow efficiently. On line 986 there is a skeleton implementation on how to fix this issue already, but it expects the vectors to be normalized. The whole point of the derivation behind https://www.xarg.org/proof/quaternion-from-two-vectors/ was to find a way of being as efficient (and numerically stable) as possible. Normalizing the vectors for this check introduces new square roots, but it must be solved.
While writing these lines I got an idea to stick to that efficiency goal while catching this case as well. I'll look into it!

**Comment by infusion** [comment]
Okay, this idea worked out pretty well. Could you please give it a try? :)

**Comment by esromneb** [comment]
Nice! OK I tested it and my special case works! Amazing turnaround time I must say. good work!

**Comment by esromneb** [comment]
Fixed by release `1.4.1` 17a294147d6e654bc47695e2276ea58fc75df22e
